libraries/chain/controller.cpp

```diff
@@ -2738,6 +2738,46 @@ signed_block_ptr controller::fetch_block_by_id( block_id_type id )const {
    return signed_block_ptr();
 }
 
+std::shared_ptr<std::vector<char>> controller::fetch_block_stream_by_number(uint32_t block_num, bool new_protocol_version)const { try {
+
+    std::shared_ptr<std::vector<char>> v = my->blog.read_signed_block_stream_by_num(block_num);
+
+    uint32_t version = my->blog.version(); // it's preamble.version, maybe not always right?
+    bool new_blog_version = my->blog.is_new_version(version);
+
+    if ((new_blog_version && new_protocol_version) || (!new_blog_version && !new_protocol_version)) {
+        return v;
+    }
+    else{//mismatch
+        if (new_protocol_version && !new_blog_version){
+            fc::datastream<const char*> ds((*v).data(), (*v).size());
+            //unpack
+            signed_block_v0 block_v0;
+            fc::raw::unpack(ds, block_v0);
+            //convert to signed_block
+            signed_block sb(std::move(block_v0), true);
+            //pack
+            auto send_buf=std::make_shared<std::vector<char>>(sizeof(sb));
```

Comment:
You need to use `pack_size` not `sizeof`

Reply:
Fixed